src/univariate/continuous/exponential.jl

```diff
@@ -105,7 +105,7 @@ cf(d::Exponential, t::Real) = 1/(1 - t * im * scale(d))
 
 
 #### Sampling
-rand(rng::AbstractRNG, d::Exponential) = xval(d, randexp(rng))
+rand(rng::AbstractRNG, d::Exponential{T}) where {T} = xval(d, randexp(rng, T))
```

Comment:
This is a slightly different case and easily broken (e.g., when `T` is not a floating point number type). In the `TDist` and `Gamma` case we just try to avoid promotions of a sample from another `rand` call, whereas this case goes deeper into the question of how `rand` should behave wrt parameters etc. (see also https://github.com/JuliaStats/Distributions.jl/pull/1433#issuecomment-2293472385).
```suggestion
rand(rng::AbstractRNG, d::Exponential) = xval(d, randexp(rng))
```

Reply:
If we do this though, the return type of `rand(Gamma(Float32, Float32))` changes depending on the value of the shape parameter because `shape == 1` samples from `Exponential`. Should we let this happen? I imagine some people will be super surprised by such behavior.

Reply:
Hi @devmotion could you comment on this?

Reply:
I think that's an argument for using the same approach as for `Normal` here, until we move to a better/different API:
```suggestion
rand(rng::AbstractRNG, d::Exponential{T}) where {T} = xval(d, randexp(rng, float(T)))
```